search/includes/classes/class-versioningcleanupjob.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Automattic\VIP\Search;
+
+class VersioningCleanupJob {
+
+	/**
+	 * The name of the scheduled cron event to run the versioning cleanup.
+	 */
+	const CRON_EVENT_NAME = 'vip_search_versioning_cleanup';
+
+	const WEEK_IN_SECONDS = 7 * 24 * 60 * 60;
+
+	const SEARCH_ALERT_SLACK_CHAT = '#vip-go-es-alerts';
+
+	public function __construct( $indexables, $versioning ) {
+		$this->indexables = $indexables;
+		$this->versioning = $versioning;
+	}
+
+	/**
+	 * Initialize the job class.
+	 *
+	 * @access  public
+	 */
+	public function init() {
+		add_action( self::CRON_EVENT_NAME, [ $this, 'versioning_cleanup' ] );
+
+		$this->schedule_job();
+	}
+
+	/**
+	 * Schedule class versioning job.
+	 *
+	 * Add the event name to WP cron schedule and then add the action.
+	 */
+	public function schedule_job() {
+		if ( ! wp_next_scheduled( self::CRON_EVENT_NAME ) ) {
+			wp_schedule_event( time(), 'weekly', self::CRON_EVENT_NAME );
+		}
+	}
+
+	/**
+	 * Process versioning cleanup.
+	 */
+	public function versioning_cleanup() {
+		$indexables = $this->indexables->get_all();
+
+		foreach ( $indexables as $indexable ) {
+			$inactive_versions = $this->get_inactive_versions( $indexable );
+
+			foreach ( $inactive_versions as $version ) {
+				if ( ! ( $version['active'] ?? false ) ) {
+					// double check that it is not active
+					$this->send_notification( $indexable->slug, $version['number'] );
+				}
+			}
+		}
+	}
+
+
+	/**
+	 * Retrieve inactive versions to be deleted
+	 *
+	 * @param \ElasticPress\Indexable $indexable The Indexable for which to retrieve index versions
+	 * @return array Array of inactive index versions
+	 */
+	public function get_inactive_versions( \ElasticPress\Indexable $indexable ) {
```

Comment:
Minor - the naming of this function could be more direct - on a first read, this makes me think it will just return all inactive versions. A better name would reflect that it's all inactive versions that are up for deletion, like `get_stale_inactive_versions()` or something.

Reply:
done